src/lib/expression/expression_utils.cpp

```diff
@@ -108,15 +108,13 @@ void expression_adapt_to_different_lqp(std::shared_ptr<AbstractExpression>& expr
 
 std::shared_ptr<LQPColumnExpression> expression_adapt_to_different_lqp(const LQPColumnExpression& lqp_column_expression,
                                                                        const LQPNodeMapping& node_mapping) {
-  const auto node = lqp_column_expression.column_reference.original_node();
+  const auto node = lqp_column_expression.original_node.lock();
```

Comment:
`.original_node.lock()` is now frequently present in the code. How about declaring the `LQPColumnExpression`'s member variable `const std::weak_ptr<const AbstractLQPNode> original_node;` as protected and adding `std::shared_ptr<const AbstractLQPNode> LQPColumnExpression::original_node() const { return _original_node.lock(); }` so we can call `original_node()` at these locations?

Reply:
That's the way it was before. People thinking that it was simply an accessor called `original_node()` more than once (e.g., in lqp_column_expression.cpp). By forcing them to call `lock()`, I hope to avoid that unnecessary overhead.